src/preamble.js

```diff
@@ -1714,6 +1714,12 @@ function intArrayFromString(stringy, dontAddNull, length) {
 {{{ maybeExport('intArrayFromString') }}}
 
 function intArrayToString(array) {
+  if (typeof TextDecoder !== 'undefined') {
```

Comment:
Makes sense; I'll go ahead and remove that.

As far as @juj's change, I just looked into it, and apparently there's a different function `UTF8ArrayToString` that he'd added `TextDecoder` to, which seems to do basically the same thing as `intArrayToString`. It's not immediately obvious at a glance why both of these exist, but I'll just change my code to use `UTF8ArrayToString` in this PR and leave the rest to you guys.